packages/compose/src/higher-order/flow.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * Creates a flow function.
+ *
+ * Allows to choose whether to perform left-to-right or right-to-left composition.
+ *
+ * @see https://docs-lodash.com/v4/flow/
+ *
+ * @param {boolean} reverse True if right-to-left, false for left-to-right composition.
+ */
+const baseFlow =
+	( reverse: boolean = false ) =>
+	( ...funcs: Function[] ) =>
+	( ...args: unknown[] ) => {
+		const functions = funcs.flat();
+		if ( reverse ) {
+			functions.reverse();
+		}
+		return functions.reduce(
+			( prev, func ) => [ func( ...prev ) ],
+			args
+		)[ 0 ];
+	};
+
+/**
+ * Composes multiple higher-order components into a single higher-order component. Performs left-to-right function
+ * composition, where each successive invocation is supplied the return value of the previous.
+ *
+ * This is inspired by `lodash`'s `flow` function.
```

Comment:
It would be better to include the license from `lodash` like here:

https://github.com/WordPress/gutenberg/blob/a3c753d28ddb0972311e2da0136277626bdf832e/packages/element/src/serialize.js#L1-L26

Reply:
Done in https://github.com/WordPress/gutenberg/pull/44112/commits/26a930481133a87623cdc64bf9cab3746dd31311